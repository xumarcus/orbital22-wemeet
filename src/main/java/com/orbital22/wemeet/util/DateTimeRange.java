package com.orbital22.wemeet.util;

import lombok.NonNull;

import java.time.LocalDateTime;

public interface DateTimeRange {
  @NonNull
  LocalDateTime getStartDateTime();

  @NonNull
  LocalDateTime getEndDateTime();

  default boolean contains(DateTimeRange other) {
    return getStartDateTime().isBefore(other.getStartDateTime())
        && getEndDateTime().isAfter(other.getEndDateTime());
  }
}
