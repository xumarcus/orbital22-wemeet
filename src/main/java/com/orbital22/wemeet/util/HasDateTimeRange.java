package com.orbital22.wemeet.util;

import lombok.NonNull;

import java.time.LocalDateTime;

public interface HasDateTimeRange {
  @NonNull
  LocalDateTime getStartDateTime();

  @NonNull
  LocalDateTime getEndDateTime();

  static boolean isValid(HasDateTimeRange dateTimeRange) {
    return dateTimeRange.getStartDateTime().isBefore(dateTimeRange.getEndDateTime());
  }

  default boolean containsRange(HasDateTimeRange other) {
    return getStartDateTime().isBefore(other.getStartDateTime())
        && getEndDateTime().isAfter(other.getEndDateTime());
  }
}
