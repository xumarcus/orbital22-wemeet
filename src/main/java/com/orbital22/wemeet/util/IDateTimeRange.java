package com.orbital22.wemeet.util;

import lombok.NonNull;
import org.apache.commons.lang3.Range;

import java.time.LocalDateTime;
import java.time.chrono.ChronoLocalDateTime;

import static org.apache.commons.lang3.Range.between;

public interface IDateTimeRange {
  @NonNull
  LocalDateTime getStartDateTime();

  @NonNull
  LocalDateTime getEndDateTime();

  default boolean isValid() {
    return getStartDateTime().isBefore(getEndDateTime());
  }

  default Range<ChronoLocalDateTime<?>> getChronoLocalDateTimeRange() {
    return between(getStartDateTime(), getEndDateTime());
  }
}
