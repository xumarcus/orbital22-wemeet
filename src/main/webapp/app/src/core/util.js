// In development, resp.type === 'cors'
// In production, resp.type is per normal
// TODO handle fullyAuthenticated

const ajax = (method, data) => async (uri) => {
    // Spring POST /login accepts formData by default.
    // POST /logout does not take any parameters.
    const isFormData = uri.endsWith("login");
    const resp = await fetch(uri, {
        body: isFormData ? data : JSON.stringify(data),
        headers: {
            "X-XSRF-TOKEN": csrfToken(),
            ...(isFormData ? undefined : {"Content-Type": "application/json"})
        },
        method,
        redirect: "follow",
    });
    if (resp.ok) {
        const location = resp.headers.get('Location');
        if (location != null) {
            // Fetch created resource
            return ajax('GET')(location);
        }
        return resp.json();
    } else {
        throw new Error(resp.statusText, { cause: await resp.json() });
    }
};

const csrfToken = () => document.cookie.replace(/(?:^|.*;\s*)XSRF-TOKEN\s*=\s*([^;]*).*$|^.*$/, '$1');

export default ajax;