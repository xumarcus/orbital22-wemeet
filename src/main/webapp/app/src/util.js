const ajax = (method) => (uri) => {
    const headers = {'Content-Type': 'application/json'};
    return fetch(uri, { method, headers })
        .then(resp => resp.json());
}
export default ajax;