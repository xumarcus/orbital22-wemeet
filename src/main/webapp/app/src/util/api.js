import { useEffect } from 'react';

export function ajax(method, uri) {
    return fetch(uri, { method, headers: {'Content-Type': 'application/json'} });
}

export function useAjax(method, uri, setState, callback) {
    useEffect(() => {
        const fetchData = async () => {
            const response = await ajax(method, uri);
            if (response.ok) {
                const json = await response.json();
                setState(json);
            } else {
                callback();
            }
        }
        fetchData().then();
    }, []);
}