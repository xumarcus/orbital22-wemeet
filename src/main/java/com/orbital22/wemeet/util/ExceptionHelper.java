package com.orbital22.wemeet.util;

import org.jetbrains.annotations.Nullable;
import org.springframework.data.rest.core.RepositoryConstraintViolationException;
import org.springframework.validation.Errors;
import org.springframework.validation.MapBindingResult;

import java.util.HashMap;

public interface ExceptionHelper {
  static RepositoryConstraintViolationException from(
          Object object, @Nullable String field, String errorCode) {
    return from(object, field, errorCode, null);
  }

  static RepositoryConstraintViolationException from(
          Object object, @Nullable String field, String errorCode, @Nullable String defaultMessage) {
    Errors errors = new MapBindingResult(new HashMap<>(), object.getClass().getName());
    Object[] errorArgs = new Object[] {object};

    if (field != null) {
      errors.rejectValue(field, errorCode, errorArgs, defaultMessage);
    } else {
      errors.reject(errorCode, errorArgs, defaultMessage);
    }

    return new RepositoryConstraintViolationException(errors);
  }
}
