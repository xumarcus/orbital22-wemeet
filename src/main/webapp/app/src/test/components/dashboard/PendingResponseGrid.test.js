import React from 'react'
import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import {
  PendingResponseGridInner
} from '../../../components/dashboard/PendingResponseGrid'
import { setupForSyncfusionTest } from '../../test-core/util'

const sleep = ms => new Promise(resolve => setTimeout(resolve, ms))

describe('PendingResponseGrid', () => {
  // Syncfusion
  beforeAll(setupForSyncfusionTest)

  it('should render correctly without data', () => {
    const { asFragment } = render(<PendingResponseGridInner
      dataSource={[]}
      selfUserId={null}
                                  />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render correctly with data', async () => {
    const { asFragment, getByText } = render(
      <BrowserRouter>
        <PendingResponseGridInner
          dataSource={[
            {
              rosterPlan: { id: 1, title: 'Hello' },
              owner: { id: 3, email: 'test@test.com' }
            },
            {
              rosterPlan: { id: 2, title: 'World' },
              owner: { id: 4, email: 'not-test@test.com' }
            }
          ]}
          selfUserId={3}
        />
      </BrowserRouter>
    )
    await sleep(500)

    expect(getByText('Hello')).toBeInTheDocument()
    expect(getByText('Yourself')).toBeInTheDocument()

    expect(getByText('World')).toBeInTheDocument()
    expect(getByText('not-test@test.com')).toBeInTheDocument()

    expect(asFragment()).toMatchSnapshot()
  })
})
