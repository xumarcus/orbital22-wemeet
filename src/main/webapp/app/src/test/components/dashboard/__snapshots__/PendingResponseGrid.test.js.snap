// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PendingResponseGrid should render correctly with data 1`] = `
<DocumentFragment>
  <div
    class="e-control e-grid e-lib e-gridhover e-responsive e-default e-droppable e-tooltip e-keyboard"
    data-role="grid"
    id="grid_1693809010_1"
    style="width: auto;"
    tabindex="0"
  >
    <div
      class="e-dlg-ref-element"
    />
    <div
      class="e-dlg-container e-dlg-center-center"
      style="z-index: 1000000000;"
    >
      <div
        aria-describedby="grid_1693809010_1EditAlert_dialog-content"
        aria-modal="true"
        class="e-control e-dialog e-lib e-dlg-modal e-popup e-popup-close"
        id="grid_1693809010_1EditAlert"
        role="dialog"
        style="max-height: -20px; z-index: 1000000000; width: 320px; left: 0px; top: 0px;"
      >
        <div
          class="e-dlg-content"
          id="grid_1693809010_1EditAlert_dialog-content"
        />
        <div
          class="e-footer-content"
        >
          <button
            class="e-control e-btn e-lib e-flat e-primary"
            type="button"
          >
            OK
          </button>
        </div>
      </div>
      <div
        class="e-dlg-overlay"
        style="z-index: 999999999; display: none;"
      />
    </div>
    <div
      class="e-dlg-ref-element"
    />
    <div
      class="e-dlg-container e-dlg-center-center"
      style="z-index: 1000000001;"
    >
      <div
        aria-describedby="grid_1693809010_1EditConfirm_dialog-content"
        aria-modal="true"
        class="e-control e-dialog e-lib e-dlg-modal e-popup e-popup-close"
        id="grid_1693809010_1EditConfirm"
        role="dialog"
        style="max-height: -20px; z-index: 1000000001; width: 320px; left: 0px; top: 0px;"
      >
        <div
          class="e-dlg-content"
          id="grid_1693809010_1EditConfirm_dialog-content"
        />
        <div
          class="e-footer-content"
        >
          <button
            class="e-control e-btn e-lib e-primary e-flat"
            type="button"
          >
            OK
          </button>
          <button
            class="e-control e-btn e-lib e-flat"
            type="button"
          >
            Cancel
          </button>
        </div>
      </div>
      <div
        class="e-dlg-overlay"
        style="z-index: 1000000000; display: none;"
      />
    </div>
    <div
      class="e-spinner-pane e-spin-hide"
    >
      <div
        class="e-spinner-inner"
      />
    </div>
    <div
      aria-disabled="false"
      aria-haspopup="false"
      aria-orientation="horizontal"
      class="e-control e-toolbar e-lib e-keyboard"
      id="grid_1693809010_1_toolbarItems"
      role="toolbar"
      style="height: auto; width: auto;"
      tabindex="0"
    >
      <div
        class="e-toolbar-items"
      >
        <div
          aria-disabled="true"
          class="e-toolbar-item e-overlay"
          title="Add"
        >
          <button
            aria-label="Add"
            class="e-tbar-btn e-tbtn-txt e-control e-btn e-lib"
            id="grid_1693809010_1_add"
            style="width: auto;"
            tabindex="-1"
            type="button"
          >
            <span
              class="e-btn-icon e-add e-icons e-icon-left"
            />
            <span
              class="e-tbar-btn-text"
            >
              Add
            </span>
          </button>
        </div>
        <div
          aria-disabled="true"
          class="e-toolbar-item e-overlay"
          title="Edit"
        >
          <button
            aria-label="Edit"
            class="e-tbar-btn e-tbtn-txt e-control e-btn e-lib"
            id="grid_1693809010_1_edit"
            style="width: auto;"
            tabindex="-1"
            type="button"
          >
            <span
              class="e-btn-icon e-edit e-icons e-icon-left"
            />
            <span
              class="e-tbar-btn-text"
            >
              Edit
            </span>
          </button>
        </div>
        <div
          aria-disabled="false"
          class="e-toolbar-item"
          title="Delete"
        >
          <button
            aria-label="Delete"
            class="e-tbar-btn e-tbtn-txt e-control e-btn e-lib"
            id="grid_1693809010_1_delete"
            style="width: auto;"
            tabindex="-1"
            type="button"
          >
            <span
              class="e-btn-icon e-delete e-icons e-icon-left"
            />
            <span
              class="e-tbar-btn-text"
            >
              Delete
            </span>
          </button>
        </div>
        <div
          aria-disabled="true"
          class="e-toolbar-item e-overlay"
          title="Update"
        >
          <button
            aria-label="Update"
            class="e-tbar-btn e-tbtn-txt e-control e-btn e-lib"
            id="grid_1693809010_1_update"
            style="width: auto;"
            tabindex="-1"
            type="button"
          >
            <span
              class="e-btn-icon e-update e-icons e-icon-left"
            />
            <span
              class="e-tbar-btn-text"
            >
              Update
            </span>
          </button>
        </div>
        <div
          aria-disabled="true"
          class="e-toolbar-item e-overlay"
          title="Cancel"
        >
          <button
            aria-label="Cancel"
            class="e-tbar-btn e-tbtn-txt e-control e-btn e-lib"
            id="grid_1693809010_1_cancel"
            style="width: auto;"
            tabindex="-1"
            type="button"
          >
            <span
              class="e-btn-icon e-cancel e-icons e-icon-left"
            />
            <span
              class="e-tbar-btn-text"
            >
              Cancel
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="e-gridheader e-lib e-droppable"
    >
      <div
        class="e-headercontent"
      >
        <table
          aria-colcount="3"
          aria-rowcount="2"
          cellspacing="0.25px"
          class="e-table"
          role="grid"
        >
          <colgroup
            id="grid_1693809010_1colgroup"
          >
            <col
              style="width: 120px;"
            />
            <col />
            <col />
          </colgroup>
          <thead
            role="rowgroup"
          >
            <tr
              class="e-columnheader"
              role="row"
            >
              <th
                aria-colindex="0"
                aria-colspan="1"
                aria-grabbed="false"
                aria-rowspan="1"
                aria-sort="none"
                class="e-headercell e-centeralign"
                role="columnheader"
                tabindex="-1"
              >
                <div
                  class="e-headercelldiv"
                  e-mappinguid="grid-column3"
                  style="text-align: Center;"
                >
                  <span
                    class="e-headertext"
                  >
                    ID
                  </span>
                </div>
                <div
                  class="e-sortfilterdiv e-icons"
                />
              </th>
              <th
                aria-colindex="1"
                aria-colspan="1"
                aria-grabbed="false"
                aria-rowspan="1"
                aria-sort="none"
                class="e-headercell e-centeralign"
                role="columnheader"
                tabindex="-1"
              >
                <div
                  class="e-headercelldiv"
                  e-mappinguid="grid-column4"
                  style="text-align: Center;"
                >
                  <span
                    class="e-headertext"
                  >
                    Title
                  </span>
                </div>
                <div
                  class="e-sortfilterdiv e-icons"
                />
              </th>
              <th
                aria-colindex="2"
                aria-colspan="1"
                aria-grabbed="false"
                aria-rowspan="1"
                aria-sort="none"
                class="e-headercell e-lastcell e-centeralign"
                role="columnheader"
                tabindex="-1"
              >
                <div
                  class="e-headercelldiv"
                  e-mappinguid="grid-column5"
                  style="text-align: Center;"
                >
                  <span
                    class="e-headertext"
                  >
                    From
                  </span>
                </div>
                <div
                  class="e-sortfilterdiv e-icons"
                />
              </th>
            </tr>
          </thead>
          <tbody
            class="e-hide"
            role="rowgroup"
          >
            <tr
              role="row"
            >
              <td />
              <td />
              <td />
            </tr>
          </tbody>
          <caption
            class="e-hide"
          >
            grid_1693809010_1_header_table
          </caption>
        </table>
      </div>
    </div>
    <div
      class="e-gridcontent"
    >
      <div
        aria-busy="false"
        class="e-content"
        style="height: auto; overflow-y: auto; position: relative;"
      >
        <table
          aria-colcount="3"
          aria-multiselectable="false"
          cellspacing="0.25px"
          class="e-table"
          id="grid_1693809010_1_content_table"
          role="grid"
        >
          <colgroup
            id="content-grid_1693809010_1colgroup"
          >
            <col
              style="width: 120px;"
            />
            <col />
            <col />
          </colgroup>
          <tbody
            role="rowgroup"
          >
            <tr
              aria-rowindex="0"
              class="e-row"
              data-uid="grid-row6"
              role="row"
            >
              <td
                aria-colindex="0"
                aria-label="undefined is template cell column header ID"
                class="e-rowcell e-templatecell"
                index="0"
                role="gridcell"
                style="text-align: Center;"
                tabindex="-1"
              >
                <a
                  href="/meeting/1/rank"
                >
                  1
                </a>
              </td>
              <td
                aria-colindex="1"
                aria-label="Hello column header Title"
                class="e-rowcell"
                index="0"
                role="gridcell"
                style="text-align: Center;"
                tabindex="-1"
              >
                Hello
              </td>
              <td
                aria-colindex="2"
                aria-label="undefined is template cell column header From"
                class="e-rowcell e-templatecell"
                index="0"
                role="gridcell"
                style="text-align: Center;"
                tabindex="-1"
              >
                Yourself
              </td>
            </tr>
            <tr
              aria-rowindex="1"
              class="e-row e-altrow"
              data-uid="grid-row7"
              role="row"
            >
              <td
                aria-colindex="0"
                aria-label="undefined is template cell column header ID"
                class="e-rowcell e-templatecell"
                index="1"
                role="gridcell"
                style="text-align: Center;"
                tabindex="-1"
              >
                <a
                  href="/meeting/2/rank"
                >
                  2
                </a>
              </td>
              <td
                aria-colindex="1"
                aria-label="World column header Title"
                class="e-rowcell"
                index="1"
                role="gridcell"
                style="text-align: Center;"
                tabindex="-1"
              >
                World
              </td>
              <td
                aria-colindex="2"
                aria-label="undefined is template cell column header From"
                class="e-rowcell e-templatecell"
                index="1"
                role="gridcell"
                style="text-align: Center;"
                tabindex="-1"
              >
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
                  to="mailto:not-test@test.com"
                >
                  not-test@test.com
                </a>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="e-gridpopup"
      style="display:none;"
    >
      <div
        class="e-uptail e-tail"
      />
      <div
        class="e-content"
        tabindex="-1"
      >
        <span />
      </div>
      <div
        class="e-downtail e-tail"
      />
    </div>
    <textarea
      aria-label="clipboard"
      class="e-clipboard"
      style="opacity: 0"
      tabindex="-1"
    />
  </div>
</DocumentFragment>
`;

exports[`PendingResponseGrid should render correctly without data 1`] = `
<DocumentFragment>
  <div
    class="e-control e-grid e-lib e-gridhover e-responsive e-default e-droppable e-tooltip e-keyboard"
    data-role="grid"
    id="grid_1693809010_0"
    style="width: auto;"
    tabindex="0"
  >
    <div
      class="e-dlg-ref-element"
    />
    <div
      class="e-dlg-container e-dlg-center-center"
      style="z-index: 1000000000;"
    >
      <div
        aria-describedby="grid_1693809010_0EditAlert_dialog-content"
        aria-modal="true"
        class="e-control e-dialog e-lib e-dlg-modal e-popup e-popup-close"
        id="grid_1693809010_0EditAlert"
        role="dialog"
        style="max-height: -20px; z-index: 1000000000; width: 320px; left: 0px; top: 0px;"
      >
        <div
          class="e-dlg-content"
          id="grid_1693809010_0EditAlert_dialog-content"
        />
        <div
          class="e-footer-content"
        >
          <button
            class="e-control e-btn e-lib e-flat e-primary"
            type="button"
          >
            OK
          </button>
        </div>
      </div>
      <div
        class="e-dlg-overlay"
        style="z-index: 999999999; display: none;"
      />
    </div>
    <div
      class="e-dlg-ref-element"
    />
    <div
      class="e-dlg-container e-dlg-center-center"
      style="z-index: 1000000001;"
    >
      <div
        aria-describedby="grid_1693809010_0EditConfirm_dialog-content"
        aria-modal="true"
        class="e-control e-dialog e-lib e-dlg-modal e-popup e-popup-close"
        id="grid_1693809010_0EditConfirm"
        role="dialog"
        style="max-height: -20px; z-index: 1000000001; width: 320px; left: 0px; top: 0px;"
      >
        <div
          class="e-dlg-content"
          id="grid_1693809010_0EditConfirm_dialog-content"
        />
        <div
          class="e-footer-content"
        >
          <button
            class="e-control e-btn e-lib e-primary e-flat"
            type="button"
          >
            OK
          </button>
          <button
            class="e-control e-btn e-lib e-flat"
            type="button"
          >
            Cancel
          </button>
        </div>
      </div>
      <div
        class="e-dlg-overlay"
        style="z-index: 1000000000; display: none;"
      />
    </div>
    <div
      class="e-spinner-pane e-spin-hide"
    >
      <div
        class="e-spinner-inner"
      />
    </div>
    <div
      aria-disabled="false"
      aria-haspopup="false"
      aria-orientation="horizontal"
      class="e-control e-toolbar e-lib e-keyboard"
      id="grid_1693809010_0_toolbarItems"
      role="toolbar"
      style="height: auto; width: auto;"
      tabindex="0"
    >
      <div
        class="e-toolbar-items"
      >
        <div
          aria-disabled="true"
          class="e-toolbar-item e-overlay"
          title="Add"
        >
          <button
            aria-label="Add"
            class="e-tbar-btn e-tbtn-txt e-control e-btn e-lib"
            id="grid_1693809010_0_add"
            style="width: auto;"
            tabindex="-1"
            type="button"
          >
            <span
              class="e-btn-icon e-add e-icons e-icon-left"
            />
            <span
              class="e-tbar-btn-text"
            >
              Add
            </span>
          </button>
        </div>
        <div
          aria-disabled="true"
          class="e-toolbar-item e-overlay"
          title="Edit"
        >
          <button
            aria-label="Edit"
            class="e-tbar-btn e-tbtn-txt e-control e-btn e-lib"
            id="grid_1693809010_0_edit"
            style="width: auto;"
            tabindex="-1"
            type="button"
          >
            <span
              class="e-btn-icon e-edit e-icons e-icon-left"
            />
            <span
              class="e-tbar-btn-text"
            >
              Edit
            </span>
          </button>
        </div>
        <div
          aria-disabled="true"
          class="e-toolbar-item e-overlay"
          title="Delete"
        >
          <button
            aria-label="Delete"
            class="e-tbar-btn e-tbtn-txt e-control e-btn e-lib"
            id="grid_1693809010_0_delete"
            style="width: auto;"
            tabindex="-1"
            type="button"
          >
            <span
              class="e-btn-icon e-delete e-icons e-icon-left"
            />
            <span
              class="e-tbar-btn-text"
            >
              Delete
            </span>
          </button>
        </div>
        <div
          aria-disabled="true"
          class="e-toolbar-item e-overlay"
          title="Update"
        >
          <button
            aria-label="Update"
            class="e-tbar-btn e-tbtn-txt e-control e-btn e-lib"
            id="grid_1693809010_0_update"
            style="width: auto;"
            tabindex="-1"
            type="button"
          >
            <span
              class="e-btn-icon e-update e-icons e-icon-left"
            />
            <span
              class="e-tbar-btn-text"
            >
              Update
            </span>
          </button>
        </div>
        <div
          aria-disabled="true"
          class="e-toolbar-item e-overlay"
          title="Cancel"
        >
          <button
            aria-label="Cancel"
            class="e-tbar-btn e-tbtn-txt e-control e-btn e-lib"
            id="grid_1693809010_0_cancel"
            style="width: auto;"
            tabindex="-1"
            type="button"
          >
            <span
              class="e-btn-icon e-cancel e-icons e-icon-left"
            />
            <span
              class="e-tbar-btn-text"
            >
              Cancel
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="e-gridheader e-lib e-droppable"
    >
      <div
        class="e-headercontent"
      >
        <table
          aria-colcount="3"
          cellspacing="0.25px"
          class="e-table"
          role="grid"
        >
          <colgroup
            id="grid_1693809010_0colgroup"
          >
            <col
              style="width: 120px;"
            />
            <col />
            <col />
          </colgroup>
          <thead
            role="rowgroup"
          >
            <tr
              class="e-columnheader"
              role="row"
            >
              <th
                aria-colindex="0"
                aria-colspan="1"
                aria-grabbed="false"
                aria-rowspan="1"
                aria-sort="none"
                class="e-headercell e-centeralign"
                role="columnheader"
                tabindex="-1"
              >
                <div
                  class="e-headercelldiv"
                  e-mappinguid="grid-column0"
                  style="text-align: Center;"
                >
                  <span
                    class="e-headertext"
                  >
                    ID
                  </span>
                </div>
                <div
                  class="e-sortfilterdiv e-icons"
                />
              </th>
              <th
                aria-colindex="1"
                aria-colspan="1"
                aria-grabbed="false"
                aria-rowspan="1"
                aria-sort="none"
                class="e-headercell e-centeralign"
                role="columnheader"
                tabindex="-1"
              >
                <div
                  class="e-headercelldiv"
                  e-mappinguid="grid-column1"
                  style="text-align: Center;"
                >
                  <span
                    class="e-headertext"
                  >
                    Title
                  </span>
                </div>
                <div
                  class="e-sortfilterdiv e-icons"
                />
              </th>
              <th
                aria-colindex="2"
                aria-colspan="1"
                aria-grabbed="false"
                aria-rowspan="1"
                aria-sort="none"
                class="e-headercell e-lastcell e-centeralign"
                role="columnheader"
                tabindex="-1"
              >
                <div
                  class="e-headercelldiv"
                  e-mappinguid="grid-column2"
                  style="text-align: Center;"
                >
                  <span
                    class="e-headertext"
                  >
                    From
                  </span>
                </div>
                <div
                  class="e-sortfilterdiv e-icons"
                />
              </th>
            </tr>
          </thead>
          <tbody
            class="e-hide"
            role="rowgroup"
          >
            <tr
              role="row"
            >
              <td />
              <td />
              <td />
            </tr>
          </tbody>
          <caption
            class="e-hide"
          >
            grid_1693809010_0_header_table
          </caption>
        </table>
      </div>
    </div>
    <div
      class="e-gridcontent"
    >
      <div
        aria-busy="true"
        class="e-content"
        style="height: auto; overflow-y: auto;"
      >
        <table
          aria-multiselectable="false"
          cellspacing="0.25px"
          class="e-table"
          id="grid_1693809010_0_content_table"
          role="grid"
        >
          <colgroup
            id="content-grid_1693809010_0colgroup"
          >
            <col
              style="width: 120px;"
            />
            <col />
            <col />
          </colgroup>
          <tbody
            role="rowgroup"
          >
            <tr
              class="e-emptyrow"
              role="row"
            >
              <td
                colspan="3"
              >
                No records to display
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="e-gridpopup"
      style="display:none;"
    >
      <div
        class="e-uptail e-tail"
      />
      <div
        class="e-content"
        tabindex="-1"
      >
        <span />
      </div>
      <div
        class="e-downtail e-tail"
      />
    </div>
  </div>
</DocumentFragment>
`;
